### PR TITLE
Refactoring the GetGroupKeyResponse::merge function

### PR DESCRIPTION
**Pull request by inetic**

### CODE REVIEW

File: src/histogram.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+use std::collections::BTreeMap;
+
+pub struct Histogram<K: Ord + Clone> {
```
Comment (benjaminbollen) [inline src/histogram.rs]: Nice one Peter, strictly speaking a histogram collects items in defined buckets; so is not a histogram in the strict sense of the word.  Hence I called it 'frequency count' which arguably is not a good name either.  Nice call regardless of semantics to abstract these functions !

Comment (inetic) [inline src/histogram.rs]: Cheers Ben :)

**Comment by dirvine** [comment]
@chandraprakash I am happy if you are @rossmuir to

**Comment by chandraprakash** [comment]
May be worth having a test along with this PR.

**Comment by benjaminbollen** [comment]
Starting to implement tests on this PR

**Comment by chandraprakash** [comment]
@dirvine  we agreed on merging it. Travis shows failure because of nightly I think.

**Pull request merged**